rubin_sim/maf/metrics/cosmology_summary_metrics.py

```diff
@@ -0,0 +1,985 @@
+
+            mean_z_values_use = mzvals[use_bins]
+            sctz_use = sctz[use_bins]
+
+            Clbias = poly_fit(mean_z_values_use)
+            rescale_fac = sctz_use / figure_9_mean_z_scatter
+            Clbias *= rescale_fac
+            fit_res_bias = np.polyfit(mean_z_values_use, Clbias, 1)
+            poly_fit_bias = np.poly1d(fit_res_bias)
+
+            clbiasvals = poly_fit_bias(mean_z_values_use)
+            return clbiasvals, mean_z_values_use
+
+        result = np.empty(1, dtype=[("name", np.str_, 20), ("y1ratio", float), ("y10ratio", float)])
+        result["name"][0] = "MultibandMeanzBiasMetric"
+
+        # Technically don't need this for now (isn't used in previous one)
+        # need to define an array of bad values for the masked pixels
+        badval_arr = np.repeat(self.badval, len(self.filter_list))
+        # converts the input recarray to an array
+        data_slice_list = [
+            badval_arr if isinstance(x, float) else x for x in data_slice["metricdata"].tolist()
+        ]
+        lengths = np.array([len(x) for x in data_slice_list])
+        # should be (nbins, npix)
+        data_slice_arr = np.asarray(data_slice_list, dtype=float).T
+        data_slice_arr[~np.isfinite(data_slice_arr)] = (
+            hp.UNSEEN
+        )  # need to work with TotalPowerMetric and healpix
+
+        # measure rms in each bin.
+        # The original metric returns an array at each slice_point (of the
+        # original slicer) -- so there is a bit of "rearrangement" that
+        # has to happen to be able to pass a np.array with right dtype
+        # (i.e. dtype = [("metricdata", float)]) to each call to
+        # the rmsMetric `run` methods.
+        rmsarray = np.array(
+            [
+                self.rmsMetric.run(np.core.records.fromrecords(x, dtype=[("metricdata", float)]))
+                for x in data_slice_arr
+            ]
+        )  # rms values
+
+        totdz = 0
+        avmeanz = 0
+        totclbias = 0
+
+        for i, filt in enumerate(self.filter_list):
+            dzdminterp, meanzinterp = compute_dzfromdm(self.zbins, filt, self.year)
+            stdz = [float(np.abs(dz)) * float(rmsarray[i]) for dz in dzdminterp]
+
+            clbias, meanz_use = compute_Clbias(meanzinterp, stdz)
+
+            totdz += [float(st**2) for st in stdz]
+            totclbias += clbias
+            avmeanz += meanzinterp
+
+        # These should be included in self rather than hard coded
+        y10_req = 0.003
+        y1_goal = 0.013
+
+        # clbiastot = np.max(clbias) # if adding doesn't work over z range -- CHECK
+        y10ratio = totclbias / y10_req
+        y1ratio = totclbias / y1_goal
+
+        result["y1ratio"] = y1ratio
+        result["y10ratio"] = y10ratio
+
+
+class UniformAreaFoMFractionMetric(BaseMetric):
+    """
+    Run as summary metric on NestedRIZExptimeExgalM5Metric.
+
+    This metric StaticProbesFoMEmulatorMetric
```

Comment:
Looks like this needs to be expanded?  Something like:

This metric uses maps of the combined RIZ exposure time and i-band depth maps (with a consistent set of area cuts) to identify potential reductions in cosmological constraining power due to substantial large-scale power in non-uniform coadds at a particular data release.  The RIZ exposure time map is used to identify whether there are residual rolling features.  If not, the metric returns 1.  If there are such features, then the region is segmented into similar-depth regions and the one with the largest cosmological constraining power is presumed to be used for science.  In that case, the metric returns the 3x2pt FoM (`StaticProbesFoMEmulatorMetric`, quantifying weak lensing and large-scale structure constraining power) for the largest of those regions, divided by that for the full region if it had been usable.

Reply:
Thank you for catching this gap Rachel! I committed and push this change.